include/dd/Package.hpp

```diff
@@ -485,6 +485,80 @@ template <class Config> class Package {
     return f;
   }
 
+  // generate general GHZ state with n qubits
+  vEdge makeGHZState(const std::size_t n) {
+    if (n > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n == 0) {
+      throw std::runtime_error{
+          "Cannot create GHZ state for 0 qubits. Please choose n > 0."};
+    }
+
+    auto leftSubtree = vEdge::one();
+    auto rightSubtree = vEdge::one();
+
+    for (std::size_t p = 0; p < n - 1; ++p) {
+      leftSubtree = makeDDNode(static_cast<Qubit>(p),
+                               std::array{leftSubtree, vEdge::zero()});
+      rightSubtree = makeDDNode(static_cast<Qubit>(p),
+                                std::array{vEdge::zero(), rightSubtree});
+    }
+
+    auto f = makeDDNode(static_cast<Qubit>(n - 1),
+                        std::array<vEdge, RADIX>{
+                            {{leftSubtree.p, cn.lookup(dd::SQRT2_2, 0)},
+                             {rightSubtree.p, cn.lookup(dd::SQRT2_2, 0)}}});
+
+    return f;
+  }
+
+  // generate general W state with n qubits
+  vEdge makeWState(const std::size_t n) {
+    if (n > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n == 0) {
+      throw std::runtime_error{
+          "Cannot create W state for 0 qubits. Please choose n > 0."};
+    }
+
+    if (n == 1) {
+      return makeBasisState(n, {BasisStates::one});
+    }
```

Comment:
If you adopt the changes below, I believe you can actually drop this special case here as it is natively handled.
```suggestion
```